docs/shared/concepts/sync-generators.md

```diff
@@ -0,0 +1,46 @@
+# Sync Generators
+
+In Nx 19.7, you can use sync generators to update code before a task is run or before code is sent to CI. Sync generators are designed to perform maintenance work that is required for a particular task or for CI.
+
+Sync Generator Examples:
+
+- Ensure code is formatted in a specific way before CI is run
+- Update a custom CI script with binning strategies based on the current project graph
+- Update TypeScript config files with project references based on the current project graph
+
+## Task Sync Generators
+
+Sync generators can be associated with a particular task. Nx will ensure that the sync generator is run before the task is executed. This is similar to the `dependsOn` property, but for generators instead of task dependencies.
```

Comment:
```suggestion
Sync generators can be associated with a particular task. Nx will ensure that the sync generator is run before the task is executed (unless `sync.applyChanges` is set to `false` in `nx.json`). This is similar to the `dependsOn` property, but for generators instead of task dependencies.
```